### DISCUSSION TITLE
Integrate Zitadel into docker with traefik

### DISCUSSION THREAD

**Discussion by JeremyKj87** [zitadel/zitadel#4832] [Q&A]
Hello,

I'm trying to deploy an application accessing data stored in an OpenSearch instance and OpenSearch Dashboards using docker and docker-compose. For routing to the corresponding containers, a traefik container is used. The configuration is, so far, based only on the PathPrefix option, so that all containers are reachable via a prefix, e.g., `localhost/dashboards` , rather than a separate domain. This keeps the deployment portable in terms of hostnames/domains. All containers are part of the same docker network, so that the required communication between the containers can rely solely on their names.

To secure the contained applications, such as Dashboards, it makes sense to use an identity provider, such as Zitadel. All involved containers do have support for OpenID, which seems like the way to go to manage users and permissions.

Currently, I have two problems.
First, it seems that Zitadel cannot be integrated with the aforementioned approach using path prefixes only. Domains are required, is this true?
Second, there seems to be a problem with the domain-based approach of Zitadel. The docker containers connect and authenticate from the internal docker network. At the same time, the Zitadel frontend should be accessible via traefik from "outside" of the docker network. Is this correct, or am I missing some configuration options that allow such an access?

Any ideas are much appreciated!

**Comment by eliobischof** [comment]
> First, it seems that Zitadel cannot be integrated with the aforementioned approach using path prefixes only. Domains are required, is this true?

If you use a single instance [virtual instance](https://zitadel.com/docs/concepts/structure/instance#multiple-virtual-instances), you can configure your apex domain for ZITADEL using the ExternalDomain config during setup. I'm not sure if all [ZITADEL paths](https://zitadel.com/docs/apis/introduction#zitadel-path-prefixes) are configurable, though. @adlerhurst can you shine some llight on this, plz?

> Second, there seems to be a problem with the domain-based approach of Zitadel. The docker containers connect and authenticate from the internal docker network. At the same time, the Zitadel frontend should be accessible via traefik from "outside" of the docker network. Is this correct, or am I missing some configuration options that allow such an access?

ZITADEL maps a requests Host header to the virtual instance. So, you could eighter just send the Host header with your internal API calls or proxy also internal requests for example with Traefik and set the host header there. 

Does that make sense?
